bin/pipefymessage

```diff
@@ -35,6 +38,9 @@ module PipefyMessage
         require File.expand_path("config/application.rb")
         require File.expand_path("config/environment.rb")
       end
+
+      # Override rails logger by gem logger
+      ::Rails.logger = logger
```

Comment:
**question:** Considering that everything is running in the same context and we add pipefy_message in publishers in the monolith, wouldn't this impact everything?

Reply:
This line is executed only when it is executed with `bundle exec pipefymessage -R`, i.e., when it is used to run the consumer worker.

When the gem is used in the publisher, it keeps using what it configured by Rails, eg:

Code in users_controller.rb: ```
  def edit
    Rails.logger.warn('===> app/controllers/users_controller.rb#edit - it remains as simple string <==')
    @menu = Menu.new(
      context: 'user-settings',
      help_context: 'user_settings'
    )
  end
```

Logs: ```
  [primary] Role Load (0.4ms)  SELECT "roles".* FROM "roles" INNER JOIN "users_roles" ON "roles"."id" = "users_roles"."role_id" WHERE "roles"."deleted_at" IS NULL AND "users_roles"."user_id" = 65 AND ((((roles.name = 'external_guest') AND (roles.resource_type IS NULL) AND (roles.resource_id IS NULL)) OR ((roles.name = 'external_guest') AND (roles.resource_type = 'Organization') AND (roles.resource_id IS NULL)) OR ((roles.name = 'external_guest') AND (roles.resource_type = 'Organization') AND (roles.resource_id = 17)))) /*application:Pipefy,controller:users,action:edit,line:(eval):41:in `block in exec_query'*/
===> app/controllers/users_controller.rb#edit - it remains as simple string <==
  Rendering users/edit.html.erb within layouts/legacy_pipestyle
```

Reply:
I think the way to go is doing it the other way around... using the app logger on the gem, to standardize everything and deal with the context thing more effectively, since the consumer is still running in the context of the app. Plus, @gabriel376 has already argued in favor of moving consumer initialization to the core rather than the gem, as this is a bit of a breach of abstraction, I guess, so it might not be a good idea to rely on this initialization for a solution.

Reply:
I agree that it would be better if the logger was something decoupled, but it doesn't seem to be something easy to do because the `Rails.logger` itself is something very coupled to the Rails framework.

If we go to the approach of using the `Rails.logger` as it is, then I guess we should give up saving our logs as json in Logfy. Otherwise a lot of logs will be ended up in logfy DLQ.

Reply:
I was looking at `Pipefy::Telemtry::Logging.Fomatter` and it should log in json format. So maybe there is a way to initialize it .

Reply:
It turns out that if we set RAILS_ENV=production before starting the consumer (`RAILS_ENV=production bundle exec pipefymessage -w FormulaFields::Consumer::CardFieldValueUpdated -R`), then Rails.logger will generate logs in json format.

```
root@app-core-consumer-formula-field-card-field-value-updated-626cl9:/pipefy# RAILS_ENV=production bundle exec pipefymessage -w FormulaFields::Consumer::CardFieldValueUpdated -R

{"time":"2022-08-17T18:03:49+00:00","level":"WARN","program_name":null,"correlation_id":"","message":"Creating scope :matching_value. Overwriting existing method CardFieldValue.matching_value."}
{"time":"2022-08-17T18:03:50+00:00","level":"WARN","program_name":null,"correlation_id":"","message":"Creating scope :open. Overwriting existing method InboxEmailEvent.open."}
{"time":"2022-08-17T18:03:50+00:00","level":"WARN","program_name":null,"correlation_id":"","message":"Creating scope :fail. Overwriting existing method MigratorTrack.fail."}
```

I am reverting my changes in `pipefymessage.rb`.

Tks for all the inputs!

Reply:
The log below was generated by a line which is inside `pipefy-core` on the consumer and sent to Logfy:

![image](https://user-images.githubusercontent.com/8394463/185236412-77af7261-c23d-480b-ae3f-96caca7a374a.png)